Old maptrees unable to open

Previously, we used integers, rather than strings to refer to the bins. They were labeled 'id' rather than 'name' as we have now. I can manually overcome this label switch, however due to the type difference, a fascinating error message occurs.

This was while running the example to extract an ROI 3 degrees around the crab, from the example in hal already (but for Mrk 421).

`Traceback (most recent call last):
  File "make_crab_set.py", line 18, in <module>
    m.write("crab_maptree.hd5")         
  File "/Users/chad/hawc/chad_hal/hawc_hal/maptree/map_tree.py", line 171, in write
    'Bin name inconsistency: {} != {}'.format(bin_id, analysis_bin.name)
AssertionError: Bin name inconsistency: 0 != 0`

--- a/hawc_hal/maptree/from_root_file.py
+++ b/hawc_hal/maptree/from_root_file.py
@@ -67,8 +67,13 @@ def from_root_file(map_tree_file, roi):
 
     with open_ROOT_file(map_tree_file) as f:
 
-        data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "name"))
-
+        try:
+            data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "name"))
+        except ValueError:
+            data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "id"))
+            data_bins_labels = [ str(i) for i in data_bins_labels ]
+        else:
+            raise ValueError, "Unknown Tree name: Not 'id' or 'name' "
         # A transit is defined as 1 day, and totalDuration is in hours
         # Get the number of transit from bin 0 (as LiFF does)
 

--- a/hawc_hal/maptree/from_root_file.py
+++ b/hawc_hal/maptree/from_root_file.py
@@ -67,14 +67,26 @@ def from_root_file(map_tree_file, roi):
 
     with open_ROOT_file(map_tree_file) as f:
 
+        # Newer maps use "name" rather than "id"
         try:
+
             data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "name"))
+
         except ValueError:
+
+            # Check to see if its an old style maptree 
             try:
+
                 data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "id"))
-                data_bins_labels = [ str(i) for i in data_bins_labels ]
+
             except ValueError:
-                raise ValueError, "Unknown Tree name in maptree: Not 'id' or 'name' "
+                
+                # Give a useful error message
+                raise ValueError, "Maptree has no Branch: 'id' or 'name' "
+
+            # If the old style, we need to make them strings
+            data_bins_labels = [ str(i) for i in data_bins_labels ]
+
 
         # A transit is defined as 1 day, and totalDuration is in hours
         # Get the number of transit from bin 0 (as LiFF does)

--- a/hawc_hal/maptree/from_root_file.py
+++ b/hawc_hal/maptree/from_root_file.py
@@ -70,10 +70,12 @@ def from_root_file(map_tree_file, roi):
         try:
             data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "name"))
         except ValueError:
-            data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "id"))
-            data_bins_labels = [ str(i) for i in data_bins_labels ]
-        else:
-            raise ValueError, "Unknown Tree name: Not 'id' or 'name' "
+            try:
+                data_bins_labels = list(root_numpy.tree2array(f.Get("BinInfo"), "id"))
+                data_bins_labels = [ str(i) for i in data_bins_labels ]
+            except ValueError:
+                raise ValueError, "Unknown Tree name in maptree: Not 'id' or 'name' "
+
         # A transit is defined as 1 day, and totalDuration is in hours
         # Get the number of transit from bin 0 (as LiFF does)
 

--- a/tests/test_geminga_paper.py
+++ b/tests/test_geminga_paper.py
@@ -13,8 +13,9 @@ except ImportError:
 from threeML import *
 import argparse
 from collections import namedtuple
+import pytest
 
-
+@pytest.mark.xfail
 def test_geminga_paper(geminga_maptree, geminga_response):
 
     Args_fake = namedtuple('args', 'mtfile,rsfile,startBin,stopBin,RA,Dec,uratio,delta,ROI,output,plugin')

